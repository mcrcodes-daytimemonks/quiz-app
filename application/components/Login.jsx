import { signIn } from "next-auth/react";

const Login = () => {

  return (
    <div>
      <p>Hey there stranger!
      <br />
      <br />
      Please sign in so that we can get to know you better!</p>
      <form onSubmit={signIn}>
        <button type="submit">Sign In</button>
      </form>
    </div>
  );
};

export default Login;
