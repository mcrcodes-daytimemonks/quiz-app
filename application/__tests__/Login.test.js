import { render, screen, fireEvent } from "@testing-library/react";
import Login from "../components/Login";
import "@testing-library/jest-dom";

const props = {
  username: "username",
  handleUsernameChange: jest.fn(),
  handleLogin: jest.fn(),
};

const renderLogin = () => render(<Login {...props} />);

describe("Login", () => {
  it("matches snapshot", () => {
    const { asFragment } = renderLogin();

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a textbox for the user to enter their name", () => {
    renderLogin();

    const usernameInput = screen.getByRole("textbox", {
      name: /enter your name/i,
    });

    expect(usernameInput).toHaveAttribute("name", "username");
    expect(usernameInput).toHaveAttribute("id", "username");
    expect(usernameInput).toHaveAttribute("type", "text");
    expect(usernameInput).toHaveAttribute("maxLength", "35");
    expect(usernameInput).toHaveValue(props.username);
  });

  it("renders a login button", () => {
    renderLogin();

    expect(screen.getByRole("button", { name: /login/i })).toBeInTheDocument();
  });
});
