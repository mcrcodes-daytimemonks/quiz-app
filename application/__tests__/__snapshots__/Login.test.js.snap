// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <p>
      Hey there stranger!
      <br />
      <br />
      Please sign in so that we can get to know you better!
    </p>
    <form>
      <button
        type="submit"
      >
        Sign In
      </button>
    </form>
  </div>
</DocumentFragment>
`;
